src/context.rs

```diff
@@ -54,6 +70,82 @@ pub struct RunningState {
 }
 
 impl Context {
+    pub fn new(cb: Box<ContextCallback>, os_name: String, dbfile: PathBuf) -> Result<Context> {
+        let mut blob_fname = OsString::new();
+        blob_fname.push(dbfile.file_name().unwrap_or_default());
+        blob_fname.push("-blobs");
+        let blobdir = dbfile.with_file_name(blob_fname);
+        if !blobdir.exists() {
+            std::fs::create_dir_all(&blobdir)?;
+        }
+        Context::with_blobdir(cb, os_name, dbfile, blobdir)
+    }
+
+    pub fn with_blobdir(
+        cb: Box<ContextCallback>,
+        os_name: String,
+        dbfile: PathBuf,
+        blobdir: PathBuf,
+    ) -> Result<Context> {
+        if !blobdir.is_dir() {
+            return Err(format_err!("Blobdir does not exist: {}", blobdir.display()));
+        }
+        let blobdir_c = blobdir.to_c_string()?;
+        let ctx = Context {
+            blobdir: Arc::new(RwLock::new(unsafe { dc_strdup(blobdir_c.as_ptr()) })),
```

Comment:
why not switch the blobdir to a `PathBuf` as well?

Reply:
Yes, that's an obvious next refactor.  I was trying to keep the "unrelated" changes as small as possible though, that's why I jumped through hoops like this (e.g. sql.open() signature was another hard one to resist iirc)